ccl/changefeedccl: TestChangefeedRandomExpressions failed

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/14cc3559-c723-415f-9136-7f3e7efb765a?testReportRun=5&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-25.1.7-rc @ [8e7ab9c1c4ec161768dd3c43133200e6039503c1](https://github.com/cockroachdb/cockroach/commits/8e7ab9c1c4ec161768dd3c43133200e6039503c1):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-16237):::INT8, 147647629:::INT8, (-1729717921862528462):::INT8, 3.4028234663852886e+38:::FLOAT8, 0.0:::FLOAT8, '1971-05-04':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, true, 52776951811.56719971:::DECIMAL, e'X<\b\\':::STRING, '\xec6c0c0233c1':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '16.32.61.167/11':::INET, 'true':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (26463:::INT8, 184085213:::INT8, (-8697207185582371821):::INT8, 1.401298464324817e-45:::FLOAT8, 0.3406727040108559:::FLOAT8, NULL, '2019-12-24 20:18:32.000633':::TIMESTAMP, '2018-06-17 04:50:12.000671+00':::TIMESTAMPTZ, '1 day':::INTERVAL, true, 0:::DECIMAL, 'X':::STRING, '\xe76c911c9a':::BYTES, '16ff3850-0e02-43f7-9e7c-cbdb5e0489c8':::UUID, '251.54.12.6/26':::INET, '{"a": "ti6KEu", "b": null, "c": [null], "pUJvy9o4B": null, "sxni36l": 2.9674480287302067}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 2147483647:::INT8, (-3788997392535502752):::INT8, (-0.7597370147705078):::FLOAT8, 1.401298464324817e-45:::FLOAT8, '4714-11-24 BC':::DATE, '2019-05-06 15:37:05.000123':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '5 years 1 mon 332 days 23:08:29.015089':::INTERVAL, true, 'NaN':::DECIMAL, e'\x16qw3x':::STRING, '\x58':::BYTES, '971bf9e1-66a7-4dee-bc5f-ba4c4996a87c':::UUID, '27e:d5aa:1fab:a516:ad75:7be8:bc45:cd34/107':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-28741):::INT8, 127:::INT8, (-3413950966378918394):::INT8, 3.4028234663852886e+38:::FLOAT8, '-Inf':::FLOAT8, '1970-01-01':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 766027502334.637802:::DECIMAL, 'q':::STRING, '\x00':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'cac5:8bba:96e8:f814:3469:745c:65c1:54a3/127':::INET, '{"a": [{"a": false, "foo": "8NOTOEBk5"}, [0.1107714946906361, 0.4858331193014481], 0.15475192321185413], "foo": [["foo", null, false], {"foobar": null}, 4.038420989463418, "XK2nZqKH", false], "nFibu": 0.9696085723461201}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, (-2118680813):::INT8, 2866190820003850591:::INT8, (-0.09992647171020508):::FLOAT8, 1.0:::FLOAT8, '-infinity':::DATE, NULL, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, false, 0:::DECIMAL, e'E\fUv\nT\x01B':::STRING, NULL, '00000000-0000-0000-0000-000000000000':::UUID, '104.193.15.152/6':::INET, '[{"cvtktPoVv": 0.26176744799877455, "uCW": true}, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-1):::INT8, (-5822189156605985415):::INT8, (-0.5270695090293884):::FLOAT8, (-0.8027701765168285):::FLOAT8, '4714-11-24 BC':::DATE, '2015-01-28 18:01:33.000083':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '1 day':::INTERVAL, true, (-59691763204943610.52):::DECIMAL, e'\'':::STRING, '\xfa58ee396ec21c09':::BYTES, '1fd1825a-9d52-491d-844f-1d7127bbc5f2':::UUID, '71ce:37db:4382:eb5c:a5b5:6f58:bb0d:b9cc/103':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 2147483647:::INT8, 6853959471518069682:::INT8, NULL, NULL, '5874897-12-31':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 0:::DECIMAL, e'\U00002603':::STRING, '\x8403f1':::BYTES, '14801964-b8a8-4374-b0e1-0a60a2c52872':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 55 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._string AS "cOl'_34")
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._string AS "cOl'_34") to emit 100 events
    changefeed_test.go:1292: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string NOT LIKE seed._string
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string NOT LIKE seed._string to emit 1 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._inet::INET << seed._inet::INET)::BOOL AND false)
    changefeed_test.go:1285: Skipping predicate ((seed._inet::INET << seed._inet::INET)::BOOL AND false) because it returned no rows
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1285: Skipping predicate false because it returned no rows
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 40 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e'\n':::STRING NOT LIKE (NULL::STRING || array_position(ARRAY[B'0000001000010010100100100001000100010011111110001', B'1100100101100110101100111010001011011110101110001']::VARBIT[], B'01100101010100101110101000001010001110011000010100000111111111'::VARBIT, ((-3498172376777898265):::INT8::INT8 << NULL::INT8)::INT8::INT8)::INT8::INT8)::STRING
    changefeed_test.go:1285: Skipping predicate e'\n':::STRING NOT LIKE (NULL::STRING || array_position(ARRAY[B'0000001000010010100100100001000100010011111110001', B'1100100101100110101100111010001011011110101110001']::VARBIT[], B'01100101010100101110101000001010001110011000010100000111111111'::VARBIT, ((-3498172376777898265):::INT8::INT8 << NULL::INT8)::INT8::INT8)::INT8::INT8)::STRING because it returned no rows
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._string::STRING % seed._string::STRING)::BOOL OR (((NOT seed._bool) AND EXISTS (SELECT regrole(seed._int8::INT8)::REGROLE AS col_143 FROM d.public.seed AS "tAb_42" WHERE seed._string LIKE seed._string ORDER BY seed._bool ASC)) OR seed._bool))
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((seed._string::STRING % seed._string::STRING)::BOOL OR (((NOT seed._bool) AND EXISTS (SELECT regrole(seed._int8::INT8)::REGROLE AS col_143 FROM d.public.seed AS "tAb_42" WHERE seed._string LIKE seed._string ORDER BY seed._bool ASC)) OR seed._bool)) to emit 94 events
    changefeed_test.go:1292: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._string::STRING % seed._string::STRING)::BOOL
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._string::STRING % seed._string::STRING)::BOOL to emit 51 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string ~ seed._string
    changefeed_test.go:1278: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string ~ seed._string because error pq: invalid regular expression: error parsing regexp: missing closing ): ``no(>]��BE``
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_isclosed('010700008003000000010400008005000000010100008098973E1B82A3ED4168AA01799D2BD3C17CCAF7D38CFBF041010100008062C323AE1396004200B157A84D3A8AC1604376C80ABAB0C101010000805065994C57D5ED41380E8787A6BEF7C124B727D9E915E9C10101000080CC0B3535B8C4FE41A85486BAFC2EF841AA25AD3C15CEF64101010000800478C8FA4ADEFB4194E57E4DFB9BFC415417568847C1EDC10106000080080000000103000080010000000A0000004E4466DF2F10FEC11EA617F9D63BE1C1FC0F73B58359FD41FEE65D966108F54109877D9BF10FFEC1F28E0FA3A1D5F5C1403D6F86EAE0A04100D3750C7F859D41BCF1097340C1EAC16CBDC9A9086C01425AF7257A41FFF5419069DEFEB3B8CBC1E8F32F63E7E2D041324A61886FF00142886FB360C6B9D3C17805C8B3D77CEBC19ABF0574FD1DF9417A3C454850B9F141120900CF2005FDC1E8FF310F29B2E9416F7E5BEC8A7E01C203E555EF909BFDC1DCD84DB006C7E64182E8CC032704F0414214EBA0F96B00C21CDF16E541C2E3410069A6C2CE3BBBC14E4466DF2F10FEC11EA617F9D63BE1C1FC0F73B58359FD41010300008001000000070000008EFB6463626FFDC182D01F91AE4C02C244D02D5880F8F041209799A52D7CBB41E420C48F07FCDFC1903B529AF446FE416CF7A506298F004250B479B807B8FC4132F66D2E8815F6416E560AE153D8004218B54EEA66FC00426CC63790B8C4DEC14E60C0C1B5D0F9C1DAD6C1753A63F741D4A4F7D24C63E841A0449494B02302C2489E09D78AF3EC4150A9A200FA97CA418EFB6463626FFDC182D01F91AE4C02C244D02D5880F8F0410103000080010000000900000058F6CF8DEC0A02C2EE80C6E6F732EDC1237A8287A915F4C198A61E66EF83F5C16BF0CE4B5ABFFCC18036509B7ED2D3418C4AFD6578B5DAC1781B1F2DE41FF4C1881A1CB0D363DF41A05F45F06B92BE412058E908A7C7D941A89B155511B8FEC16849E2A6D84FDF4194388BFF5E69F541C0128498D297BB41883364CEC5FEE141907D55739265FE41F0135BBEECB501C2A030094B85C2EBC1E8F43CEE18D4FB41ABD7A5CEB875F4C16311383FB31702C2D2E9E3DEB259F241C0ABC6D37372D24158F6CF8DEC0A02C2EE80C6E6F732EDC1237A8287A915F4C101030000800100000007000000204679AAEC9DED41A83EA4A2EB1500C26844FADC629BEBC1FC98820175F9EF41F68107854553F9C1484796A646CCF241CE7F5E12333502424C3907BD10B20142FAE04B68236702C21CF4690D2AA0EF4126AF98371CD0F141DEF750788E9201C2209BED51E5B4DEC14C8639A9DEDAE24180EB3FE43775CAC106BD0B63DF7EE7C13896C8F52529DE41F8DF94CE2987D541204679AAEC9DED41A83EA4A2EB1500C26844FADC629BEBC101030000800100000007000000806886633986A84140B82CB71A87AC41204C5D9A6930DDC116615E1B327EFBC1C41B85FAA83C01C26CDB4FF916A2DFC1C05CED0BB709D64194A552E6166FDAC15A114BF2EA84F1C174C36FCEEE3E02428A3DF06CC89CF84105DDC11DB1ED01C29A25E1D90B9BF641762B8656247CF741D8D97B73E91BD4C13050F1D4D1A2D841F885F74E4E1DDA41682280BBE4A9D341806886633986A84140B82CB71A87AC41204C5D9A6930DDC10103000080010000000400000028DE4BD12392FE4173570B74102AF1C10C287F89A8BDF9C1088E850FD430F64108B5380DF54AD141C608050E19A8E6C18E9E8EA555A2E4C1A81AD848EC89FA41204E6EEA221DE74128DE4BD12392FE4173570B74102AF1C10C287F89A8BDF9C1010300008001000000080000002E7423232006FFC1D46691BF4144ECC128328557D00FF9C1F8B4DE6AA38EDA4140661B774062FBC1E09F329E8D77FBC11A32598252A0024292B438B8ADADFA41D4F9ECF16555E941909929FE78FCFE41C087A6674F82FB4190A74D6D70D1D04144E205D55895E7418AA73CF1E8A100425A6F2A5796D3EDC1CE911634D706F2C124D9575B0654FB4178FBCE86F75EF6414CA9897A55A0FBC11EFB4EA0414FF941A0F937EBB572FA412E7423232006FFC1D46691BF4144ECC128328557D00FF9C10103000080010000000400000078D7FD028091E24170378B41C25EF6C1D4B8DEA63AFBE8412C6BAFA2B8F30142B281967DE029FBC13F8596AC7FD9F4C1E83D6639D818EFC1C40A7D7041DAF14100EEB6E0BFFBFD4178D7FD028091E24170378B41C25EF6C1D4B8DEA63AFBE8410106000080050000000103000080010000000500000095CDCE0C53B900C27ABC442C7BB2E7C1CC6D5EE1AC12004266800709645AF2C1C2A92A695D93E9C1549DB5A53A2FFDC156E28E8A3299F2C16C90372D59DCED417C5E6D2CDED3F7C1BD31757D06CDFAC188FD98C8A20EFB41C213B0CF39CDF04195CDCE0C53B900C27ABC442C7BB2E7C1CC6D5EE1AC1200420103000080010000000700000064293F37F2A2DEC112238D42AFACFDC1582ACAF2ED96FB41789177054F7CD74130138B7D4BCBD8C1608B7F5B5ABDDEC148F4DBE51210FD41B0766705FA05F241C2DAE9CF6AC8F0C180E1156D928EC3C1E0BAA9978732BF41C844B9E9CC9EEC4170252AF1A760E1C1907E60D9F217CD410E9C702574EFF9C1622E6770048CE5C15025374644F8D141F0AC54DB33EBEB4164293F37F2A2DEC112238D42AFACFDC1582ACAF2ED96FB410103000080010000000D0000007ECCFD01A2DAF7C1A0C6A4067136D8414EFB2B4557D7F541761221FA2B0EFAC1A093C4B99A24BE41A67096587488F3412DD4A1A67CA300C2B8E93D97542500C200BE9AD547C7F441209D15ADAA5AC0C1A03E6CE14DBAECC18EB1846CDADBF241606DF17BC596C5417DA6759DA0E3F5C14DA6D1DC5DD601C2F85BF927ABCCFC4132CB8AB6B624E0C158C25EE9864AD44120862E44E78BFC41C0E773EB9169E341800F69B62595ACC1085EAB509012F9416428046B8D8CF74180E4511C68E3D3C120AA2C4E2107CC41A013372D3076FF419493A11F627DF64150DBB4D38C77FBC1D62CE0361F6002427C95260CE275F941DA51B0B089D8EEC130ECAA7CBDDDEF418CB1DFC36795EDC10D6119F45B8800C2B4E1BAA6CA13FE41E8AA35AA7377FF417ECCFD01A2DAF7C1A0C6A4067136D8414EFB2B4557D7F54101030000800100000004000000B8BDF3670F09F5413C3DE1BDE0ABF4C18E6EF7E0B5E9EEC1D014B56D7896CDC1C0590BB6EC80004250D5D6A695F2C8C1F0A605F8B581C5C1420179DCEE43F241122AF7D09D3BFBC1B8BDF3670F09F5413C3DE1BDE0ABF4C18E6EF7E0B5E9EEC101030000800100000008000000EA37D4845DF7EEC118B2F1CE4329D3C18182AC1E1A1600C25EC782691133F5C124F3D5983149E0C16A7E80F3D33CE0C13CA05E51041BF8C1E7B9E47B7931FEC1F048BC0F230E01C21E42DCDDA8500042440A4EC5A60BF041FE60B84CF6C6E8C178255909C3A0F5415866659483EAEE419045D6DFEA2DEDC161E6891AFC24FDC100F5EA69BFB4AC41D027AD8D0C49ED41966F6C6B582EF3C100AC567DD5A9ABC1D07159A0DE32C5C1EA37D4845DF7EEC118B2F1CE4329D3C18182AC1E1A1600C2':::GEOMETRY::GEOMETRY)::BOOL
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE st_isclosed('010700008003000000010400008005000000010100008098973E1B82A3ED4168AA01799D2BD3C17CCAF7D38CFBF041010100008062C323AE1396004200B157A84D3A8AC1604376C80ABAB0C101010000805065994C57D5ED41380E8787A6BEF7C124B727D9E915E9C10101000080CC0B3535B8C4FE41A85486BAFC2EF841AA25AD3C15CEF64101010000800478C8FA4ADEFB4194E57E4DFB9BFC415417568847C1EDC10106000080080000000103000080010000000A0000004E4466DF2F10FEC11EA617F9D63BE1C1FC0F73B58359FD41FEE65D966108F54109877D9BF10FFEC1F28E0FA3A1D5F5C1403D6F86EAE0A04100D3750C7F859D41BCF1097340C1EAC16CBDC9A9086C01425AF7257A41FFF5419069DEFEB3B8CBC1E8F32F63E7E2D041324A61886FF00142886FB360C6B9D3C17805C8B3D77CEBC19ABF0574FD1DF9417A3C454850B9F141120900CF2005FDC1E8FF310F29B2E9416F7E5BEC8A7E01C203E555EF909BFDC1DCD84DB006C7E64182E8CC032704F0414214EBA0F96B00C21CDF16E541C2E3410069A6C2CE3BBBC14E4466DF2F10FEC11EA617F9D63BE1C1FC0F73B58359FD41010300008001000000070000008EFB6463626FFDC182D01F91AE4C02C244D02D5880F8F041209799A52D7CBB41E420C48F07FCDFC1903B529AF446FE416CF7A506298F004250B479B807B8FC4132F66D2E8815F6416E560AE153D8004218B54EEA66FC00426CC63790B8C4DEC14E60C0C1B5D0F9C1DAD6C1753A63F741D4A4F7D24C63E841A0449494B02302C2489E09D78AF3EC4150A9A200FA97CA418EFB6463626FFDC182D01F91AE4C02C244D02D5880F8F0410103000080010000000900000058F6CF8DEC0A02C2EE80C6E6F732EDC1237A8287A915F4C198A61E66EF83F5C16BF0CE4B5ABFFCC18036509B7ED2D3418C4AFD6578B5DAC1781B1F2DE41FF4C1881A1CB0D363DF41A05F45F06B92BE412058E908A7C7D941A89B155511B8FEC16849E2A6D84FDF4194388BFF5E69F541C0128498D297BB41883364CEC5FEE141907D55739265FE41F0135BBEECB501C2A030094B85C2EBC1E8F43CEE18D4FB41ABD7A5CEB875F4C16311383FB31702C2D2E9E3DEB259F241C0ABC6D37372D24158F6CF8DEC0A02C2EE80C6E6F732EDC1237A8287A915F4C101030000800100000007000000204679AAEC9DED41A83EA4A2EB1500C26844FADC629BEBC1FC98820175F9EF41F68107854553F9C1484796A646CCF241CE7F5E12333502424C3907BD10B20142FAE04B68236702C21CF4690D2AA0EF4126AF98371CD0F141DEF750788E9201C2209BED51E5B4DEC14C8639A9DEDAE24180EB3FE43775CAC106BD0B63DF7EE7C13896C8F52529DE41F8DF94CE2987D541204679AAEC9DED41A83EA4A2EB1500C26844FADC629BEBC101030000800100000007000000806886633986A84140B82CB71A87AC41204C5D9A6930DDC116615E1B327EFBC1C41B85FAA83C01C26CDB4FF916A2DFC1C05CED0BB709D64194A552E6166FDAC15A114BF2EA84F1C174C36FCEEE3E02428A3DF06CC89CF84105DDC11DB1ED01C29A25E1D90B9BF641762B8656247CF741D8D97B73E91BD4C13050F1D4D1A2D841F885F74E4E1DDA41682280BBE4A9D341806886633986A84140B82CB71A87AC41204C5D9A6930DDC10103000080010000000400000028DE4BD12392FE4173570B74102AF1C10C287F89A8BDF9C1088E850FD430F64108B5380DF54AD141C608050E19A8E6C18E9E8EA555A2E4C1A81AD848EC89FA41204E6EEA221DE74128DE4BD12392FE4173570B74102AF1C10C287F89A8BDF9C1010300008001000000080000002E7423232006FFC1D46691BF4144ECC128328557D00FF9C1F8B4DE6AA38EDA4140661B774062FBC1E09F329E8D77FBC11A32598252A0024292B438B8ADADFA41D4F9ECF16555E941909929FE78FCFE41C087A6674F82FB4190A74D6D70D1D04144E205D55895E7418AA73CF1E8A100425A6F2A5796D3EDC1CE911634D706F2C124D9575B0654FB4178FBCE86F75EF6414CA9897A55A0FBC11EFB4EA0414FF941A0F937EBB572FA412E7423232006FFC1D46691BF4144ECC128328557D00FF9C10103000080010000000400000078D7FD028091E24170378B41C25EF6C1D4B8DEA63AFBE8412C6BAFA2B8F30142B281967DE029FBC13F8596AC7FD9F4C1E83D6639D818EFC1C40A7D7041DAF14100EEB6E0BFFBFD4178D7FD028091E24170378B41C25EF6C1D4B8DEA63AFBE8410106000080050000000103000080010000000500000095CDCE0C53B900C27ABC442C7BB2E7C1CC6D5EE1AC12004266800709645AF2C1C2A92A695D93E9C1549DB5A53A2FFDC156E28E8A3299F2C16C90372D59DCED417C5E6D2CDED3F7C1BD31757D06CDFAC188FD98C8A20EFB41C213B0CF39CDF04195CDCE0C53B900C27ABC442C7BB2E7C1CC6D5EE1AC1200420103000080010000000700000064293F37F2A2DEC112238D42AFACFDC1582ACAF2ED96FB41789177054F7CD74130138B7D4BCBD8C1608B7F5B5ABDDEC148F4DBE51210FD41B0766705FA05F241C2DAE9CF6AC8F0C180E1156D928EC3C1E0BAA9978732BF41C844B9E9CC9EEC4170252AF1A760E1C1907E60D9F217CD410E9C702574EFF9C1622E6770048CE5C15025374644F8D141F0AC54DB33EBEB4164293F37F2A2DEC112238D42AFACFDC1582ACAF2ED96FB410103000080010000000D0000007ECCFD01A2DAF7C1A0C6A4067136D8414EFB2B4557D7F541761221FA2B0EFAC1A093C4B99A24BE41A67096587488F3412DD4A1A67CA300C2B8E93D97542500C200BE9AD547C7F441209D15ADAA5AC0C1A03E6CE14DBAECC18EB1846CDADBF241606DF17BC596C5417DA6759DA0E3F5C14DA6D1DC5DD601C2F85BF927ABCCFC4132CB8AB6B624E0C158C25EE9864AD44120862E44E78BFC41C0E773EB9169E341800F69B62595ACC1085EAB509012F9416428046B8D8CF74180E4511C68E3D3C120AA2C4E2107CC41A013372D3076FF419493A11F627DF64150DBB4D38C77FBC1D62CE0361F6002427C95260CE275F941DA51B0B089D8EEC130ECAA7CBDDDEF418CB1DFC36795EDC10D6119F45B8800C2B4E1BAA6CA13FE41E8AA35AA7377FF417ECCFD01A2DAF7C1A0C6A4067136D8414EFB2B4557D7F54101030000800100000004000000B8BDF3670F09F5413C3DE1BDE0ABF4C18E6EF7E0B5E9EEC1D014B56D7896CDC1C0590BB6EC80004250D5D6A695F2C8C1F0A605F8B581C5C1420179DCEE43F241122AF7D09D3BFBC1B8BDF3670F09F5413C3DE1BDE0ABF4C18E6EF7E0B5E9EEC101030000800100000008000000EA37D4845DF7EEC118B2F1CE4329D3C18182AC1E1A1600C25EC782691133F5C124F3D5983149E0C16A7E80F3D33CE0C13CA05E51041BF8C1E7B9E47B7931FEC1F048BC0F230E01C21E42DCDDA8500042440A4EC5A60BF041FE60B84CF6C6E8C178255909C3A0F5415866659483EAEE419045D6DFEA2DEDC161E6891AFC24FDC100F5EA69BFB4AC41D027AD8D0C49ED41966F6C6B582EF3C100AC567DD5A9ABC1D07159A0DE32C5C1EA37D4845DF7EEC118B2F1CE4329D3C18182AC1E1A1600C2':::GEOMETRY::GEOMETRY)::BOOL to emit 100 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE inet_same_family(inet(e'\x13(,jq4a\x05':::STRING::STRING)::INET::INET, '76.11.224.192/13':::INET::INET)::BOOL
    changefeed_test.go:1278: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE inet_same_family(inet(e'\x13(,jq4a\x05':::STRING::STRING)::INET::INET, '76.11.224.192/13':::INET::INET)::BOOL because error pq: inet(): could not parse "\x13(,jq4a\x05" as inet. invalid IP
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_intersects('0107000060E6100000030000000106000060E6100000080000000103000040010000000900000094EB393A737F48C0706DDF27EFFB2C40E8FE00C15907D4418CAF3DBA603D5D402C115BC38F4555C0A8C34C17FBD2F841BCDCFF75707260402CD00C238BE134C02C5034B30A56FD41CE1EBDCC330A6440201E57C309204C403FD3DEAD2024F7C1A88B04159AE76040947E298A3D014B407E4BFB9A6789EBC108D50083E80165402A3AE0C445885140E619E171928E0242F0861D68A8E24B40B88D289D33D14A40CA950EFD2772E6C1FB163256DCCF65C0A27796951389504028ACB5C44A21DD4194EB393A737F48C0706DDF27EFFB2C40E8FE00C15907D441010300004001000000050000001493636406B865400C6920A7ABF54AC01EE7FB78494300425033B30DDA7C5640006B00668B7D1F400A4F56F4550F0042BAD5CD2D48CE4EC08C243DA2A7AF4C4056310841FFEE01C20CCCEF6E494D61C02E2833A8384D5240E89F93A7F71FD6C11493636406B865400C6920A7ABF54AC01EE7FB784943004201030000400100000009000000788DB15ACC1A47C0A62D6E57AA4255C02E7AD2080B24FAC1C0245C73E55F314010AA69558C6043C0AC939729B8ABF7C1FA349A38B374664050FC2E2D242A2BC02C94ECA12970F5413C90D97D19D7604040793665C494F43F5C0CB6BA3A3EED418040163BB9762B40B2A8BD28441C504008B3E411E897E9414E65B3408E5651C04073070251FB5440E4265FA20D5101C22B54374B546555C088AC940DF8DF5540159F53EADD1BFFC141C06841CF725CC0408AE4EC5250454078E7FB472E86E541788DB15ACC1A47C0A62D6E57AA4255C02E7AD2080B24FAC10103000040010000000A000000047A57E04A6161C0EBDFB0B487CA4CC0048FF29FA857E5C110939C799E32594095026781F01850C0EA33B968C701E5C1909523A39F0C5E40682C634BCCAE2440C24C8FE09CBBF441561E0458EC12624026C188F09CA642404B05F1ECA8B9FCC12C8EDE5E25BF5740F82E3CD5FB2A42407FC15503E1B6F0C1CC1052038A1765405012FCB7781A544088F0C1D3D256FE41B81051E27C7A524020B5DB393B9446400613200D076BE1C1B0D41E41EC8362C05A9056FB8BA751403E1EE9D0AB12F0418E0F4DBFDF075FC02891678164D84C4024B32FB09EE2E941047A57E04A6161C0EBDFB0B487CA4CC0048FF29FA857E5C10103000040010000000D00000028325E494C5763C058783F2BEF0A39C0E99F8720BDA3F3C1F57DE3E8978256C081381F24EDD247C0D280470A0876F6C170B6997B26022CC07F9655B7AC964EC0EC7851A4CE99F641E00FC423D52014C020C19CC6148629C068C980D7577AF1C1E04958C51DF85740875AF5AE484452C0BC0D828FC7BEE9415A7A59DF7664654040AAD9513A5D374039A91C422BBA00C27C40857B01ED58401E44CBE29F51564032CF3C3D2377E2C100282B302B721D40D86679F643E94D40504A6092A634DB41204B9495C5F749C008E48D5DA07E53403E6E8B4EBEA400C2702D3D750D6957C0882BE82D91B141408B66332045D701C2D9396350860163C0B41FCF1A662D3E40E08A8141860ADC414CC092E1B53260C018B8668EE8112540D28A2BD4734A004228325E494C5763C058783F2BEF0A39C0E99F8720BDA3F3C10103000040010000000A0000003841DD9CD24F3CC04CFF384E435C40C0400C9D13BABEA4C1C41F597714705F4086EE5E8702714EC0C0B483190316B0410833616249276340F433010DFF8B3540E8A43585A8A5FAC1006C8CC125AE5A400C0821BDFF464940B8FDAD4C9E14D64170A63EF609FC6540AC0772FAD17752402C8C99EA33D7E2410C3BD997ABA147C0F6BE2AFA6FB65140823BD6418D3D00C2DA0FBF3C195B62C00645263D1FD651406E972EDDF9BFF6417C8BCCB832944EC0943F7B8B9F7D48408060D4417A86D7414E9D4794260E5AC074C2319416134A40476F26557AFBFCC13841DD9CD24F3CC04CFF384E435C40C0400C9D13BABEA4C10103000040010000000D000000985E0D976EE762C09DBE5DCC79C144C00652F38ADA0EF841402779B2FA0351C0E49A44AA1BD336C07EE3699648A8F641E03E1BF3A45E36C0FAD0D274CEF24CC0D83573840228E0418C88880FC3BE5040F5A22D0C997841C058A3EBE066A1D041888A7ADA43C260406C493071D80A3AC0B49F26408302FB41FCD10775D7105E4000A2E706E880054060189CF66A92DBC1C856BD85FF215340A041B65C21D62D401281FB604C91F741BEE05C4798C26340C4CB0625C3EF53402ECAA35FB289ECC1F0327008CFF73040784A0780779F5540B0891DB250D8D3C1506BC70D8EB140C090BBD2C0E64E4D402079422E932EB9C16E21F39C52424AC01407A0E2C5CB4B40BF263342F67FF2C1E4F8FB76ACC662C0AC91EF41E8333C40D09CE2F81CFBC441985E0D976EE762C09DBE5DCC79C144C00652F38ADA0EF84101030000400100000004000000A60309A0DB5B57C0F86B6A7F0F524BC07C33095513170042FA2860E2728E6040A5C3FE4ABF9B4CC0E8549E3BE54CE1C104CA164F7FE8534090FB63E74B332CC0A031C8381783C9C1A60309A0DB5B57C0F86B6A7F0F524BC07C330955131700420102000060E61000000500000050FA46AF730C5C4030599957597115C0C63F08440C03F0C1D8CBA41B080059408CFB8F4745EB4640005A7D8ACF6C8741AC6DD36A2BAA554006B9B01989C2544040E69BCD9FE5F74180F13DD791D20840BE19D5FA47DE50407C3224C50774F5C193595476192151C04C322B8C4C5336406CD5C309D530F4410105000060E610000005000000010200004003000000C440E444D0F657406C25115E2D76384070CD281EC15FC3C19EEEC2674A6F624084B9A7B9887C4B40F0D82F21C796DB41A824B89870B25340B0D027B1782D5340A6B1A9DD2B4EF8C101020000400200000010965BEA0FC23440C69B53C0767D55C092B3F62F2269E3C1507AC1825AF35B401ED72CFE12E33CC0A0FE5A68DE35CC4101020000400600000014AF1B8846B162C0100A8A2996CF55C038CE3F9FDC3AED41AA86AAC10C2857C0CE29F4C23D5555C00A5DFA3DA363F541DC8473AC2D684040A64901031EDA52C04C977DBAA0A6F9C1001C370FD0F819C0A47B4F821BFF32C0D03363D3FD87F9417ACC95D5ADB64EC0C83DD55309374440C4687DAAB8EBEB4110C9DDEDD64063C04405383A0BD738C02A661018545FE3C1010200004004000000CC2269A968B44A40A11CFD9832E44AC0185FCA555AA90042AE3468B44F69614044D9ADB273B53A40A088D66845B2D1C1507EFB71756D5440286FE31E4086464016AC1999B156F541080CAD01A5F56040E424FA1E98585140B4A56A16FEF4FB41010200004002000000008CAA6019E12240A84D72FA099B53C086427C4BE391F6C12094572524F92D402FC3EB01034350C0C6CDAF8832AB0142':::GEOGRAPHY::GEOGRAPHY, '0106000060E61000000100000001030000400100000004000000507AFDA73B7341C011E7558F9AB34EC06E3A7C95343AEEC1AC7913A7C22F6440C20037B27E2450409829DD2772A200C268E446F844483940B87C58441EB54940E42F1C7BC8A3FDC1507AFDA73B7341C011E7558F9AB34EC06E3A7C95343AEEC1':::GEOGRAPHY::GEOGRAPHY)::BOOL
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE _st_intersects('0107000060E6100000030000000106000060E6100000080000000103000040010000000900000094EB393A737F48C0706DDF27EFFB2C40E8FE00C15907D4418CAF3DBA603D5D402C115BC38F4555C0A8C34C17FBD2F841BCDCFF75707260402CD00C238BE134C02C5034B30A56FD41CE1EBDCC330A6440201E57C309204C403FD3DEAD2024F7C1A88B04159AE76040947E298A3D014B407E4BFB9A6789EBC108D50083E80165402A3AE0C445885140E619E171928E0242F0861D68A8E24B40B88D289D33D14A40CA950EFD2772E6C1FB163256DCCF65C0A27796951389504028ACB5C44A21DD4194EB393A737F48C0706DDF27EFFB2C40E8FE00C15907D441010300004001000000050000001493636406B865400C6920A7ABF54AC01EE7FB78494300425033B30DDA7C5640006B00668B7D1F400A4F56F4550F0042BAD5CD2D48CE4EC08C243DA2A7AF4C4056310841FFEE01C20CCCEF6E494D61C02E2833A8384D5240E89F93A7F71FD6C11493636406B865400C6920A7ABF54AC01EE7FB784943004201030000400100000009000000788DB15ACC1A47C0A62D6E57AA4255C02E7AD2080B24FAC1C0245C73E55F314010AA69558C6043C0AC939729B8ABF7C1FA349A38B374664050FC2E2D242A2BC02C94ECA12970F5413C90D97D19D7604040793665C494F43F5C0CB6BA3A3EED418040163BB9762B40B2A8BD28441C504008B3E411E897E9414E65B3408E5651C04073070251FB5440E4265FA20D5101C22B54374B546555C088AC940DF8DF5540159F53EADD1BFFC141C06841CF725CC0408AE4EC5250454078E7FB472E86E541788DB15ACC1A47C0A62D6E57AA4255C02E7AD2080B24FAC10103000040010000000A000000047A57E04A6161C0EBDFB0B487CA4CC0048FF29FA857E5C110939C799E32594095026781F01850C0EA33B968C701E5C1909523A39F0C5E40682C634BCCAE2440C24C8FE09CBBF441561E0458EC12624026C188F09CA642404B05F1ECA8B9FCC12C8EDE5E25BF5740F82E3CD5FB2A42407FC15503E1B6F0C1CC1052038A1765405012FCB7781A544088F0C1D3D256FE41B81051E27C7A524020B5DB393B9446400613200D076BE1C1B0D41E41EC8362C05A9056FB8BA751403E1EE9D0AB12F0418E0F4DBFDF075FC02891678164D84C4024B32FB09EE2E941047A57E04A6161C0EBDFB0B487CA4CC0048FF29FA857E5C10103000040010000000D00000028325E494C5763C058783F2BEF0A39C0E99F8720BDA3F3C1F57DE3E8978256C081381F24EDD247C0D280470A0876F6C170B6997B26022CC07F9655B7AC964EC0EC7851A4CE99F641E00FC423D52014C020C19CC6148629C068C980D7577AF1C1E04958C51DF85740875AF5AE484452C0BC0D828FC7BEE9415A7A59DF7664654040AAD9513A5D374039A91C422BBA00C27C40857B01ED58401E44CBE29F51564032CF3C3D2377E2C100282B302B721D40D86679F643E94D40504A6092A634DB41204B9495C5F749C008E48D5DA07E53403E6E8B4EBEA400C2702D3D750D6957C0882BE82D91B141408B66332045D701C2D9396350860163C0B41FCF1A662D3E40E08A8141860ADC414CC092E1B53260C018B8668EE8112540D28A2BD4734A004228325E494C5763C058783F2BEF0A39C0E99F8720BDA3F3C10103000040010000000A0000003841DD9CD24F3CC04CFF384E435C40C0400C9D13BABEA4C1C41F597714705F4086EE5E8702714EC0C0B483190316B0410833616249276340F433010DFF8B3540E8A43585A8A5FAC1006C8CC125AE5A400C0821BDFF464940B8FDAD4C9E14D64170A63EF609FC6540AC0772FAD17752402C8C99EA33D7E2410C3BD997ABA147C0F6BE2AFA6FB65140823BD6418D3D00C2DA0FBF3C195B62C00645263D1FD651406E972EDDF9BFF6417C8BCCB832944EC0943F7B8B9F7D48408060D4417A86D7414E9D4794260E5AC074C2319416134A40476F26557AFBFCC13841DD9CD24F3CC04CFF384E435C40C0400C9D13BABEA4C10103000040010000000D000000985E0D976EE762C09DBE5DCC79C144C00652F38ADA0EF841402779B2FA0351C0E49A44AA1BD336C07EE3699648A8F641E03E1BF3A45E36C0FAD0D274CEF24CC0D83573840228E0418C88880FC3BE5040F5A22D0C997841C058A3EBE066A1D041888A7ADA43C260406C493071D80A3AC0B49F26408302FB41FCD10775D7105E4000A2E706E880054060189CF66A92DBC1C856BD85FF215340A041B65C21D62D401281FB604C91F741BEE05C4798C26340C4CB0625C3EF53402ECAA35FB289ECC1F0327008CFF73040784A0780779F5540B0891DB250D8D3C1506BC70D8EB140C090BBD2C0E64E4D402079422E932EB9C16E21F39C52424AC01407A0E2C5CB4B40BF263342F67FF2C1E4F8FB76ACC662C0AC91EF41E8333C40D09CE2F81CFBC441985E0D976EE762C09DBE5DCC79C144C00652F38ADA0EF84101030000400100000004000000A60309A0DB5B57C0F86B6A7F0F524BC07C33095513170042FA2860E2728E6040A5C3FE4ABF9B4CC0E8549E3BE54CE1C104CA164F7FE8534090FB63E74B332CC0A031C8381783C9C1A60309A0DB5B57C0F86B6A7F0F524BC07C330955131700420102000060E61000000500000050FA46AF730C5C4030599957597115C0C63F08440C03F0C1D8CBA41B080059408CFB8F4745EB4640005A7D8ACF6C8741AC6DD36A2BAA554006B9B01989C2544040E69BCD9FE5F74180F13DD791D20840BE19D5FA47DE50407C3224C50774F5C193595476192151C04C322B8C4C5336406CD5C309D530F4410105000060E610000005000000010200004003000000C440E444D0F657406C25115E2D76384070CD281EC15FC3C19EEEC2674A6F624084B9A7B9887C4B40F0D82F21C796DB41A824B89870B25340B0D027B1782D5340A6B1A9DD2B4EF8C101020000400200000010965BEA0FC23440C69B53C0767D55C092B3F62F2269E3C1507AC1825AF35B401ED72CFE12E33CC0A0FE5A68DE35CC4101020000400600000014AF1B8846B162C0100A8A2996CF55C038CE3F9FDC3AED41AA86AAC10C2857C0CE29F4C23D5555C00A5DFA3DA363F541DC8473AC2D684040A64901031EDA52C04C977DBAA0A6F9C1001C370FD0F819C0A47B4F821BFF32C0D03363D3FD87F9417ACC95D5ADB64EC0C83DD55309374440C4687DAAB8EBEB4110C9DDEDD64063C04405383A0BD738C02A661018545FE3C1010200004004000000CC2269A968B44A40A11CFD9832E44AC0185FCA555AA90042AE3468B44F69614044D9ADB273B53A40A088D66845B2D1C1507EFB71756D5440286FE31E4086464016AC1999B156F541080CAD01A5F56040E424FA1E98585140B4A56A16FEF4FB41010200004002000000008CAA6019E12240A84D72FA099B53C086427C4BE391F6C12094572524F92D402FC3EB01034350C0C6CDAF8832AB0142':::GEOGRAPHY::GEOGRAPHY, '0106000060E61000000100000001030000400100000004000000507AFDA73B7341C011E7558F9AB34EC06E3A7C95343AEEC1AC7913A7C22F6440C20037B27E2450409829DD2772A200C268E446F844483940B87C58441EB54940E42F1C7BC8A3FDC1507AFDA73B7341C011E7558F9AB34EC06E3A7C95343AEEC1':::GEOGRAPHY::GEOGRAPHY)::BOOL to emit 100 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool OR (NULL AND seed._bool))
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool OR (NULL AND seed._bool)) to emit 55 events
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (15.81777815587794415:::DECIMAL IS NOT DISTINCT FROM seed.crdb_internal_origin_timestamp)
    changefeed_test.go:1285: Skipping predicate (15.81777815587794415:::DECIMAL IS NOT DISTINCT FROM seed.crdb_internal_origin_timestamp) because it returned no rows
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (false AND (seed._bool OR seed._bool))
    changefeed_test.go:1285: Skipping predicate (false AND (seed._bool OR seed._bool)) because it returned no rows
    changefeed_test.go:1273: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '':::STRING NOT ILIKE (seed._string::STRING || statement_timestamp():::TIMESTAMP::TIMESTAMP::TIMESTAMP)::STRING
    changefeed_test.go:1289: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE '':::STRING NOT ILIKE (seed._string::STRING || statement_timestamp():::TIMESTAMP::TIMESTAMP::TIMESTAMP)::STRING to emit 92 events
    changefeed_test.go:1340: operation "kafka.Next-1077055623893581825" timed out after 30.001s (given timeout 30s): context deadline exceeded
    --- FAIL: TestChangefeedRandomExpressions/kafka (42.16s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #147527 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.2 release-blocker]
- #147247 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-2 T-cdc branch-master]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51137

--- a/pkg/ccl/changefeedccl/cdceval/validation_test.go
+++ b/pkg/ccl/changefeedccl/cdceval/validation_test.go
@@ -206,6 +206,29 @@ func TestNormalizeAndValidate(t *testing.T) {
 			stmt:      "SELECT *, cdc_prev() FROM foo AS bar",
 			expectErr: `unknown function: cdc_prev()`,
 		},
+		{ // Regression for #147573: type-annotated CDC timestamp functions should
+			// not produce double type annotations after normalization.
+			name:       "statement_timestamp type annotation",
+			desc:       fooDesc,
+			stmt:       "SELECT *, statement_timestamp():::TIMESTAMP FROM foo",
+			expectStmt: "SELECT *, statement_timestamp():::TIMESTAMP FROM foo",
+		},
+		{ // Regression for #147573: event_op has no ambiguous return type, but
+			// the shared cdcFnProps pointer was mutated by cdcTimestampBuiltin
+			// functions (which have preferred overloads), causing event_op to
+			// incorrectly get AmbiguousReturnType and a spurious :::STRING
+			// annotation.
+			name:       "event_op no spurious type annotation",
+			desc:       fooDesc,
+			stmt:       "SELECT *, event_op() FROM foo",
+			expectStmt: "SELECT *, event_op() FROM foo",
+		},
+		{ // Regression for #147573: same as above but with TIMESTAMPTZ.
+			name:       "statement_timestamp timestamptz annotation",
+			desc:       fooDesc,
+			stmt:       "SELECT *, statement_timestamp():::TIMESTAMPTZ FROM foo",
+			expectStmt: "SELECT *, statement_timestamp():::TIMESTAMPTZ FROM foo",
+		},
 		{ // Regression for #115245
 			name:      "changefeed_creation_timestamp",
 			desc:      fooDesc,
@@ -248,8 +271,21 @@ func TestNormalizeAndValidate(t *testing.T) {
 			require.Equal(t, tc.expectStmt, serialized)
 
 			// Make sure we can deserialize back.
-			_, err = ParseChangefeedExpression(serialized)
+			sc2, err := ParseChangefeedExpression(serialized)
 			require.NoError(t, err)
+
+			// Verify that re-normalizing the deserialized expression produces
+			// the same result (idempotent round-trip). This catches bugs where
+			// normalization adds spurious type annotations that accumulate on
+			// each round-trip.
+			norm2, _, _, err := normalizeAndPlan(
+				ctx, &execCfg, username.RootUserName(), defaultDBSessionData,
+				tc.desc, schemaTS, target, sc2, tc.splitColFams,
+			)
+			require.NoError(t, err)
+			serialized2 := AsStringUnredacted(norm2)
+			require.Equal(t, serialized, serialized2,
+				"expression is not stable across round-trips")
 		})
 	}
 }

--- a/pkg/sql/sem/tree/expr.go
+++ b/pkg/sql/sem/tree/expr.go
@@ -1646,6 +1646,15 @@ func (node *AnnotateTypeExpr) Format(ctx *FmtCtx) {
 				return
 			}
 		}
+		// Similarly, FuncExpr.Format adds its own :::TYPE annotation when
+		// FmtParsable is set and the function has AmbiguousReturnType. Skip
+		// the AnnotateTypeExpr annotation to prevent double type annotation,
+		// which can cause type-check errors on re-parse.
+		if funcExpr, ok := node.Expr.(*FuncExpr); ok {
+			if ctx.HasFlags(FmtParsable) && funcExpr.typ != nil && funcExpr.fnProps != nil && funcExpr.fnProps.AmbiguousReturnType {
+				return
+			}
+		}
 		ctx.WriteString(":::")
 		ctx.FormatTypeReference(node.Type)
 

--- a/pkg/sql/sem/tree/function_definition.go
+++ b/pkg/sql/sem/tree/function_definition.go
@@ -174,22 +174,24 @@ func NewFunctionDefinition(
 ) *FunctionDefinition {
 	overloads := make([]*Overload, len(def))
 
+	// Copy props to avoid mutating the caller's pointer.
+	propsCopy := *props
 	for i := range def {
 		if def[i].OverloadPreference == OverloadPreferencePreferred {
 			// Builtins with a preferred overload are always ambiguous.
-			props.AmbiguousReturnType = true
+			propsCopy.AmbiguousReturnType = true
 			break
 		}
 	}
 
 	for i := range def {
-		def[i].FunctionProperties = *props
+		def[i].FunctionProperties = propsCopy
 		overloads[i] = &def[i]
 	}
 	return &FunctionDefinition{
 		Name:               name,
 		Definition:         overloads,
-		FunctionProperties: *props,
+		FunctionProperties: propsCopy,
 	}
 }
 

